src/Compilers/CSharp/Portable/Symbols/Wrapped/WrappedTypeParameterSymbol.cs

```diff
@@ -77,6 +77,14 @@ public override bool HasReferenceTypeConstraint
             }
         }
 
+        public override bool IsReferenceTypeFromConstraintTypes
+        {
+            get
+            {
+                return _underlyingTypeParameter.IsReferenceTypeFromConstraintTypes || CalculateIsReferenceTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics);
```

Comment:
>|| CalculateIsReferenceTypeFromConstraintTypes(ConstraintTypesNoUseSiteDiagnostics) [](start = 83, length = 83)

Is this necessary?

Reply:
> Is this necessary? Yes, due to possible type substitution involved new types can appear in the constraint types

---
In reply to: [521607603](https://github.com/dotnet/roslyn/pull/49261#discussion_r521607603) [](ancestors = 521607603)